xpublish/rest.py

```diff
@@ -217,8 +227,15 @@ def to_dict(data: bool = False):
 
         @self._app.get("/{var}/{chunk}")
         def get_key(var, chunk):
-            result = self.get_key(var, chunk)
-            return result
+            # First check that this request wasn't for variable metadata
+            if array_meta_key in chunk:
+                return self.zmetadata["metadata"][f"{var}/{array_meta_key}"]
+            elif attrs_key in chunk:
+                return self.zmetadata["metadata"][f"{var}/{attrs_key}"]
+            elif group_meta_key in chunk:
+                raise HTTPException(status_code=404, detail="No subgroups")
+            else:
+                return self.get_key(var, chunk)
```

Comment:
I wish this was easier. Ideally we could have separate routes for the the `.zarray` and `.zattrs` keys. @lsetiawan - do you know if it would work to define routes like:

```python
        @self._app.get("/{var}/.zarray")
...
        @self._app.get("/{var}/.zattrs")
...
        @self._app.get("/{var}/{chunk}")
...
```

In my initial test, everything was getting routed through the most general key `"/{var}/{chunk}"`.